### PR TITLE
[script.skin.helper.service] 1.0.90

### PR DESCRIPTION
**Pull request by marcelveldt**
update to 1.0.90 - quick follow up update with mostly optimizations.

Thanks in advance for updating!

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
I think i'm going to start complaining about this add-on. It has grown fat and it's just a huge pile of whatever can be found. Start slimming it down.

**Comment by marcelveldt** [comment]
@MartijnKaijser I know, I know. I already discussed this with @phil65 a couple of weeks ago.
Altough all features in the script can be enabled/controlled by the skinner seperately it's become pretty huge.

I will start breaking it up into multiple smaller (module) addons within a few weeks.
For now I'd like to ask a little patience so I have the time to do the splitting up in the next couple of weeks. So, can this be merged onto the repo untill then ?

About your other request...  Yes ,I'd love to help out adding the usefull parts to Kodi itself. Most of the stuff in this skinhelper script are just workarounds for stuff that is not yet available in Kodi core. I'd love to contibute adding this to the Kodi core but I'm not sure if I'm skilled enough to make those changes in C++ but I can sure give it a try.

Yes, you're right about the JSON changes for Emby. I've not been really involved a lot lately in the Emby development due to the fact that I'm pretty busy with my skin and skinhelper.
A few months back one of teh Emby devs checked out the Kodi core code and he tried to extend the JSON RPC features to add content to the database which worked pretty well but it was slow. I'm not sure if he spent some more time in the meanwhile to it. The whole idea was that the Emby team could do a PR to the Kodi team for the necessary changes to make the sync happen between Emby server and Kodi as client.

I hope you understand that I do my best to help out where I can to extend Kodi and it's great community. Atm I have no skilled experience with C++ so I tried adding some features with the python addon.

**Comment by marcelveldt** [comment]
any news on this, do you want me to help out with something ?

File: script.skin.helper.service/resources/lib/ArtworkUtils.py [hunk 1]
```diff
@@ -1240,11 +1241,23 @@ def getCustomFolderPath(path, foldername):
...
         if pathfound: break
     return pathfound
+
+def getSongDurationString(seconds):
+    sec = timedelta(seconds=int(seconds))
+    d = datetime(1,1,1) + sec
+    if d.second < 10:
+        secstr = "0%d" %d.second
+    else:
+        secstr = str(d.second)
```
Comment (phil65) [inline script.skin.helper.service/resources/lib/ArtworkUtils.py]: have a look at  zfill()

File: script.skin.helper.service/resources/lib/Utils.py [hunk 2]
```diff
@@ -64,12 +64,14 @@ def getContentPath(libPath):
...
-        if "ActivateWindow(MusicLibrary," in libPath:
-            libPath = libPath.replace("ActivateWindow(MusicLibrary," ,"musicdb://").lower()
+    if "activate" in libPath.lower():
+        if "activatewindow(musiclibrary," in libPath.lower():
+            libPath = libPath.lower().replace("activatewindow(musiclibrary," ,"musicdb://")
             libPath = libPath.replace(",return","/")
             libPath = libPath.replace(", return","/")
         else:
+            libPath = libPath.lower().replace(",return","")
+            libPath = libPath.lower().replace(", return","")
```
Comment (phil65) [inline script.skin.helper.service/resources/lib/Utils.py]: two times .lower() ?

**Pull request merged**